src/charm.py

```diff
@@ -162,6 +162,10 @@ def _on_update_status(self, event):
             return
 
         container = self.unit.get_container(self.name)
+        valid_pebble_plan = self._validate_pebble_plan(container)
+        if not valid_pebble_plan:
+            self.unit.status = BlockedStatus("Missing/incomplete pebble plan.")
```

Comment:
Fair comment on the `BlockedStatus` vs `MaintenanceStatus`, based on [these definitions](https://juju.is/docs/juju/status) perhaps the `WaitingStatus` is the most appropriate? I will update to this.

In terms of no pebble plan being found, this was due to the `on_update_status` hook sending the charm into an error state, in an error state all operations are frozen so the hook that reapplies the pebble plan `on_superset_pebble_ready` was not able to be executed. This PR resolves this error thereby allowing the pebble plan to be reapplied on restart.

Reply:
I've added a line `self._update(event)` to directly execute the re-application of the pebble plan from within this update status hook, this will make sure that in the event that the original event does not succeed (due to not having all relations ready or being able to connect to the container) the event will be retried on a regular interval.